paasta_tools/instance/kubernetes.py

```diff
@@ -623,6 +654,47 @@ def get_pod_containers(pod: V1Pod) -> List[Dict[str, Any]]:
     return containers
 
 
+def get_versions_for_controller_revisions(
+    controller_revisions: Sequence[V1ControllerRevision],
+    pods: Sequence[V1Pod],
+    backends: Optional[Set[str]],
+) -> List[KubernetesVersionDict]:
+    versions = []
+
+    cr_by_shas: Dict[Tuple[str, str], V1ControllerRevision] = {}
+    for cr in controller_revisions:
+        git_sha = cr.metadata.labels["paasta.yelp.com/git_sha"]
+        config_sha = cr.metadata.labels["paasta.yelp.com/config_sha"]
+        cr_by_shas[(git_sha, config_sha)] = cr
+
+    pods_by_shas: DefaultDict[Tuple[str, str], List[V1Pod]] = defaultdict(list)
+    for pod in pods:
+        git_sha = pod.metadata.labels["paasta.yelp.com/git_sha"]
+        config_sha = pod.metadata.labels["paasta.yelp.com/config_sha"]
+        pods_by_shas[(git_sha, config_sha)].append(pod)
+
+    for (git_sha, config_sha), cr in cr_by_shas.items():
+        pods = pods_by_shas[(git_sha, config_sha)]
+        versions.append(get_version_for_controller_revision(cr, pods, backends),)
+    return versions
+
+
+def get_version_for_controller_revision(
+    cr: V1ControllerRevision, pods: Sequence[V1Pod], backends: Optional[Set[str]],
+) -> KubernetesVersionDict:
+    ready_pods = [pod for pod in pods if kubernetes_tools.is_pod_ready(pod)]
+    return {
+        "name": cr.metadata.name,
+        "type": "ControllerRevision",
```

Comment:
nit: Should this be changed to `StatefulSet`, since that and `ReplicaSet` are the listed version types in the Swagger and OAPI spec?

Reply:
oh yeah.  or maybe i should change them in the specs, honestly.  will fix before i ship